app/forms/idv/finance_form.rb

```diff
@@ -4,49 +4,70 @@ class FinanceForm
 
     FINANCE_TYPES = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
 
-    attr_reader :idv_params, :finance_type, :finance_account
+    FINANCE_HTML_OPTIONS = {
+      ccn: { maxlength: 8 }
+    }
 
-    validates :finance_type, :finance_account, presence: true
+    attr_reader :idv_params, :finance_type, *FINANCE_TYPES
 
-    validates :finance_account,
+    validates :finance_type, presence: true
+
+    FINANCE_TYPES.each do |type|
+      validates type, presence: true, if: ->(form) { form.finance_type == type }
+    end
+
+    validates :ccn,
               format: { with: /\A\d{8}\z/, message: I18n.t('idv.errors.invalid_ccn') },
               if: :ccn?
 
-    validate :finance_type_valid
+    validate :validate_finance_type
 
     def initialize(idv_params)
       @idv_params = idv_params
       self.finance_type = FINANCE_TYPES.find { |param| idv_params.key? param }
-      self.finance_account = idv_params[finance_type]
+      send("#{finance_type}=", idv_params[finance_type]) if finance_type
+      self.finance_type ||= :ccn
```

Comment:
Maybe I wasn't clear? If there is no finance_type, I want to avoid the `send("{finance_type}=", value)` part, so defaulting to `:ccn` would make it look like there is a finance type, so I want to set the default value afterwards.